src/tools/compiletest/src/header/needs.rs

```diff
@@ -84,6 +84,11 @@ pub(super) fn handle_needs(
             condition: config.has_enzyme,
             ignore_reason: "ignored when LLVM Enzyme is disabled",
         },
+        Need {
+            name: "needs-enzyme-disabled",
+            condition: !config.has_enzyme,
+            ignore_reason: "ignored when LLVM Enzyme is enabled",
+        },
```

Comment:
Suggestion: instead of a `Need`, look in `cfg.rs` where we can add e.g.

https://github.com/rust-lang/rust/blob/df1b5d3cc2117f1ee96abca25678bc5f5604d450/src/tools/compiletest/src/header/cfg.rs#L163-L167

```rs
 condition! { 
     name: "built-with-enzyme-support", 
     condition: config.has_enzyme, 
     message: "when rustc is built with LLVM Enzyme" 
 } 
```

Then this will auto-register a `//@ ignore-build-with-enzyme-support` directive.

Reply:
Oddly enough it doesn't @jieyouxu 
```rust
+++ b/src/tools/compiletest/src/header/cfg.rs
@@ -166,6 +166,12 @@ macro_rules! condition {
         message: "when the target vendor is Apple"
     }
 
+    condition! {
+        name: "build-with-enzyme-support",
+        condition: config.has_enzyme,
+        message: "when rustc is built with LLVM Enzyme"
+    }
+
```
gives:
```
Testing stage1 compiletest suite=ui mode=ui (x86_64-unknown-linux-gnu)
error: detected unknown compiletest test directive `ignore-build-with-enzyme-support` in /home/manuel/prog/rust-frontend/tests/ui/feature-gates/feature-gate-autodiff.rs:3
errors encountered during EarlyProps parsing: /home/manuel/prog/rust-frontend/tests/ui/feature-gates/feature-gate-autodiff.rs
thread 'main' panicked at src/tools/compiletest/src/header.rs:81:13:
errors encountered during EarlyProps parsing
stack backtrace:
   0: rust_begin_unwind
   1: core::panicking::panic_fmt
   2: <compiletest::header::EarlyProps>::from_file
   3: compiletest::make_test
   4: compiletest::collect_tests_from_dir
   5: compiletest::collect_tests_from_dir
   6: compiletest::make_tests
   7: compiletest::run_tests
   8: compiletest::main
```
I tried both the raw `//@ ignore-build-with-enzyme-support` as well as adding it to a revision (for which there is an example in tests/ui/abi/c-zst.rs.

Reply:
Oh sorry I forgor, you'll need to add `ignore-build-with-enzyme-support` as a "known directive" inside `command-list.rs`. I want to get rid of that eventually after reworking directive handling, but it is what it is atm.

Reply:
yep thanks, I just realized it after posting (which is why I deleted my answer a few minutes ago). It works now in one mode, I'm just rebuilding the other branch to check.